internal/exec/stages/files/files.go

```diff
@@ -182,6 +182,11 @@ func (tmp dirEntry) create(l *log.Logger, u util.Util) error {
 				return err
 			}
 		}
+
+		if err = os.Chmod(path, os.FileMode(d.Mode)); err != nil {
```

Comment:
I've added an `os.Chown` to the directories now, `fileEntry.create` performs `os.Chown` in `PerformFetch`.

I ended up having to add a utility function to get the User & Group ID's as the current method used in `dirEntry.create` only works when they're specified by `id`.